management_api_app/api/dependencies/workspaces.py

```diff
@@ -14,3 +16,21 @@ async def get_workspace_by_workspace_id_from_path(workspace_id: UUID4 = Path(...
         return workspaces_repo.get_workspace_by_workspace_id(workspace_id)
     except EntityDoesNotExist:
         raise HTTPException(status_code=HTTP_404_NOT_FOUND, detail=strings.WORKSPACE_DOES_NOT_EXIST)
+
+
+async def get_deployed_workspace_by_workspace_id_from_path(workspace_id: UUID4 = Path(...), workspaces_repo: WorkspaceRepository = Depends(get_repository(WorkspaceRepository))) -> Workspace:
+    try:
+        return workspaces_repo.get_deployed_workspace_by_workspace_id(workspace_id)
+    except EntityDoesNotExist:
+        raise HTTPException(status_code=HTTP_404_NOT_FOUND, detail=strings.WORKSPACE_DOES_NOT_EXIST)
+    except ResourceIsNotDeployed:
```

Comment:
Should this really be a 404? While the workspace is not deployed it still exists

Reply:
I looked at all 4XX codes and didn't find something that felt a better fit.  But I am up for any change you prefer here...

Reply:
Maybe a 409 Conflict